parse.go

```diff
@@ -258,16 +283,24 @@ type ParseOption func(*parseOptSet)
 // parseOptSet represents the flattened option set after all ParseOptions have been applied.
 type parseOptSet struct {
 	skipMetadataReadOnNewParserInit bool
+	allowMismatchPixelDataLength    bool
 }
 
-func toParseOptSet(opts ...ParseOption) *parseOptSet {
-	optSet := &parseOptSet{}
+func toParseOptSet(opts ...ParseOption) parseOptSet {
+	optSet := parseOptSet{}
 	for _, opt := range opts {
-		opt(optSet)
+		opt(&optSet)
 	}
 	return optSet
 }
 
+// AllowMismatchPixelDataLength allow parser not returning an error when the length calculated from elements do not match with value length
```

Comment:
please add a period to the end of the sentence [1], and made some edits as well.

```suggestion
// AllowMismatchPixelDataLength allows the parser to ignore errors when the length calculated from elements do not match with value length.
```

You may also want to explicitly spell out the behavior that the data is still parsed into an errorneous encapsulated pixeldata frame (or remove that behavior if you don't need it in your use case). 

[1] https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences